platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TemporaryFileBackedOutputStream.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import com.google.common.io.ByteSource;
+import com.google.common.io.FileBackedOutputStream;
+
+/**
+ * TemporaryFileBackedOutputStream buffers the written data to memory of a temporary file, and
+ * makes the data available as a ByteSource. This class will make sure the temporary file is
+ * deleted when {@link #close()} is called. The method {@link #asByteSource()} should not be
+ * called after {@link #close()} is called.
+ */
+public class TemporaryFileBackedOutputStream extends OutputStream {
+
+    private final FileBackedOutputStream fileBackedOutputStream;
+
+    private boolean isClosed = false;
+
+    /**
+     * @param fileThreshold the number of bytes before the stream should switch to buffering to a file
+     */
+    public TemporaryFileBackedOutputStream(int fileThreshold) {
```

Comment:
It might be clearer what 1000000 is used for if it is moved to this class, and used as a constant default fileThreshold value when no arguments are passed to the constructor.